sources/System/Bonus/PlayerBonusSystem.cpp

```diff
@@ -17,9 +21,20 @@ namespace BBM
 		auto &playerBonus = entity.get<PlayerBonusComponent>();
 
 		playerBonus.nextNoClipRate -= dtime;
-		if (playerBonus.nextNoClipRate <= 0ns) {
+		if (playerBonus.nextNoClipRate <= 0ns && playerBonus.isNoClipOn) {
 			playerBonus.nextNoClipRate = playerBonus.noClipBonusRate;
 			playerBonus.isNoClipOn = false;
+			auto playerPos = entity->tryGetComponent<PositionComponent>();
+			auto playerHealth = entity->tryGetComponent<HealthComponent>();
+			if (!playerHealth || !playerPos)
+				return;
+			for (auto &[other, pos, _] : this->_wal.getScene()->view<PositionComponent, CollisionComponent>()) {
+				if (other.hasComponent<TagComponent<Player>>())
+					continue;
+				auto vec = playerPos->position.abs() - pos.position.abs();
```

Comment:
I believe you want to do get the distance between the two vectors. To do so, you should use (playerPos->position - pos.position).abs(). Your formula is wrong.

Reply:
This is why you have to use .abs() on your next line.

Reply:
(since players are always in positive, doing a position.abs() here does absolutly nothing.

Reply:
They can be negative if you're close enough to the wall.